apps/hashdotai/glossary/WIP_labeled-property-graphs.mdx

```diff
@@ -0,0 +1,65 @@
+---
+title: Labeled Property Graphs
+description: "Labeled Property Graphs consist of nodes (also called vertices) and relationships (edges), each of which can hold associated data."
+slug: labeled-property-graphs
+tags: ["Graphs"]
+---
+
+# **Data Modeling Principles**
+
+## **Structure of Labeled Property Graphs**
+
+A **labeled property graph (LPG)** consists of *entities* (also called nodes or vertices) and *links* (relationships or edges), each of which can have a series of attributes associated with it which contain relevant information. In an LPG, entities have simple textual labels like `Person` or `Product`, and so do links - for example, `Friends With` or `Purchased`. These help categorize entities and links.
+
+In more advanced "LPG inspired" systems like HASH, instead of using simple textual labels, entities and links are assigned one or more [entity types](/guide/types/entity-types) or [link types](/guide/types/link-types) (as appropriate).
+
+In LPGs, both entities and links can have any number of _properties_, which are key–value pairs storing additional information (for example, a Person node might have properties like `name:"Alice"` and `age:30`). In HASH, both entities and links can contain any number of properties, or other _links_ (allowing links to point to other links, as required).
+
+In LPGs and HASH, links are usually **directed**, meaning they have start and end at specific entities, though in some use cases direction can be ignored or traversed in both ways as needed.
+
+The LPG model is best understood with a simple example. Consider two `Person` entities and a friendship between them:
+
+```
+CREATE (p:Person { name: "Alice", age: 30 });
+CREATE (q:Person { name: "Bob", age: 32 });
+CREATE (p)-[:Friends With]->(q);
```

Comment:
Not a suggestion but just to note that in a directed graph, this example means that Alice is friends with Bob but Bob is not known to be friends with Alice (which is a possible real-world situation!)